src/practices/utils/DependenciesVersionEvaluationUtils.ts

```diff
@@ -8,9 +8,24 @@ export class DependenciesVersionEvaluationUtils {
       const parsedVersion = PackageInspectorBase.semverToPackageVersion(pkgsWithNewVersion[packageName]);
       if (parsedVersion) {
         for (const pkg of pkgs) {
-          if (pkg.name === packageName) {
-            if (parsedVersion[semverLevel] > pkg.lockfileVersion[semverLevel]) {
-              pkgsToUpdate.push({ name: pkg.name, newVersion: parsedVersion.value, currentVersion: pkg.lockfileVersion.value });
+          if (pkg.name === packageName && parsedVersion[semverLevel] > pkg.lockfileVersion[semverLevel]) {
+            switch (semverLevel) {
+              case SemverLevel.patch:
+                if (
+                  parsedVersion[SemverLevel.minor] === pkg.lockfileVersion[SemverLevel.minor] &&
```

Comment:
As we have one practice for checking whether minor or patch version is updated, I think `parsedVersion[SemverLevel.minor] === pkg.lockfileVersion[SemverLevel.minor]` condition is not necessary. Or? 
But I understand that maybe you look into the future when it's possible to split it.

Reply:
@vlasy

Reply:
@vlasy ?

Reply:
It is not necessary for that specific practice, but I wanted to fix the utils function in general. Some time in the future, it might be used to check for only patch.level changes and then the condition would be needed.

I don't consider this a premature optimization or violation of [YAGNI principle](https://github.com/dwmkerr/hacker-laws#yagni) but merely a fix so that `packagesToBeUpdated` function works "as expected" and more predictable (however subjective it is)

Reply:
You're right, it can be more predictable in the way you implemented it. And it's just one condition :)

Reply:
@adelkahomolova So can I merge it?

Reply:
Yes @prokopsimek, But it is still as a draft @vlasy